libyara/hash.c

```diff
@@ -238,6 +238,36 @@ YR_API void yr_hash_table_destroy(
   yr_free(table);
 }
 
+YR_API int yr_hash_table_iterate(
+    YR_HASH_TABLE* table,
+    const char* ns,
+    YR_HASH_TABLE_ITERATE_FUNC iterate_func,
+    void* data)
+{
+  int i;
+  int result;
+  YR_HASH_TABLE_ENTRY* entry;
+
+  if (table == NULL)
+    return ERROR_INTERNAL_FATAL_ERROR;
+
+  for (i = 0; i < table->size; i++)
```

Comment:
You can declare the `i` variable inside the `for` statement. I already abandoned the practice of declaring those variables at the beginning of the function. That practice was due to old Microsoft compilers that didn't support it, but fortunately those compilers are dead now.